ml-algorithms/src/test/java/org/opensearch/ml/engine/memory/MLMemoryManagerTest.java

```diff
@@ -0,0 +1,130 @@
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        memoryManager = new MLMemoryManager(client, clusterService, conversationMetaIndex);
+        doNothing().when(client).execute(any(), any(), any());
+        doNothing().when(client).update(any(), any());
+        when(client.admin()).thenReturn(adminClient);
+        when(adminClient.indices()).thenReturn(indicesAdminClient);
+        doNothing().when(indicesAdminClient).refresh(any(), any());
+        doNothing().when(conversationMetaIndex).checkAccess(any(), any());
+        when(clusterService.state()).thenReturn(clusterState);
+        when(clusterState.metadata()).thenReturn(metadata);
+        when(metadata.hasIndex(anyString())).thenReturn(true);
+        settings = Settings.builder().put("test_key", 10).build();
+        threadContext = new ThreadContext(settings);
+        when(client.threadPool()).thenReturn(threadPool);
+        when(threadPool.getThreadContext()).thenReturn(threadContext);
+    }
+
+    @Test
+    public void createConversation() {
+        ActionListener<CreateConversationResponse> actionListener = mock(ActionListener.class);
+        memoryManager.createConversation("test", "test", actionListener);
+    }
+
+    @Test
+    public void createInteraction() {
+        ActionListener<CreateInteractionResponse> actionListener = mock(ActionListener.class);
+        memoryManager.createInteraction("test", "test", "test", "test", "test", Map.of("feedback", "1"), "test", 0, actionListener);
+    }
+
+    @Test
+    public void createInteractionNullAdditionalInfo() {
+        ActionListener<CreateInteractionResponse> actionListener = mock(ActionListener.class);
+        memoryManager.createInteraction("test", "test", "test", "test", "test", null, "test", 0, actionListener);
+    }
+
+    @Test
+    public void getFinalInteractions() {
+        ActionListener<List<Interaction>> actionListener = mock(ActionListener.class);
+        memoryManager.getFinalInteractions("test", 1, actionListener);
+    }
+
+    @Test
+    public void innerGetFinalInteractions() {
+        ActionListener<List<Interaction>> actionListener = mock(ActionListener.class);
+        memoryManager.innerGetFinalInteractions("test", 1, actionListener);
+    }
+
+    @Test
+    public void getTracesIndex() {
+        ActionListener<List<Interaction>> actionListener = mock(ActionListener.class);
+        memoryManager.getTraces("test", actionListener);
+    }
+
+    @Test
+    public void getTracesNoIndex() {
+        ActionListener<List<Interaction>> actionListener = mock(ActionListener.class);
+        when(metadata.hasIndex(anyString())).thenReturn(false);
+        memoryManager.getTraces("test", actionListener);
+    }
+
+    @Test
+    public void updateInteraction() {
+        ActionListener<UpdateResponse> actionListener = mock(ActionListener.class);
+        memoryManager.updateInteraction("test", Map.of("feedback", "1"), actionListener);
```

Comment:
Add checking ? After update interaction, we should verify

Same comment for other test cases. We should verify after running the test code.

Reply:
Will leave to Xun's pr.